### PR TITLE
Remove SMS feature for send to device (#9864)

### PR DESCRIPTION
**Pull request by stephaniehobson**
## Description

Remove SMS feature for send to device 
- rename form field from  `phone-or-email` to `s2d-email` (unique name makes it easier to trace the path through the code)
- remove `legal_note_sms` and `input_label_alt` from macro
- remove sms error and thank you messages and form labeling
- remove geolocation check from helpers and javascript
- remove sms from message sets (keep message sets because not all countries get email option)
- remove sms tests
  - convert one test from sms to email
- update docs to remove sms references
- remove sms strings from ftl files
- update JS and CSS to remove sms behaviour and styles
- update JS tests

## Issue / Bugzilla link

See #9864 

## Testing

Testing values: success@example.com, failure@example.com, invalid@email.
Should still fallback to QR codes for languages without translated emails (eg. AR)

http://localhost:8000/en-US/firefox/mobile/
http://localhost:8000/en-US/firefox/mobile/get-app/
http://localhost:8000/en-US/firefox/whatsnew/all/
http://localhost:8000/en-US/firefox/79.0/whatsnew/all/
http://localhost:8000/en-US/firefox/80.0/whatsnew/all/
http://localhost:8000/en-US/firefox/81.0/whatsnew/all/
http://localhost:8000/en-US/firefox/welcome/4/
http://localhost:8000/en-US/firefox/welcome/5/
http://localhost:8000/en-US/firefox/welcome/6/
http://localhost:8000/en-US/firefox/welcome/7/

### CODE REVIEW

**Comment by stephaniehobson** [comment]
@pmac I would like to change the name of the input from `['phone-or-email']` to something else to reflect that phone is no longer acceptable. How do I do that without screwing up basket?

**Comment by stephaniehobson** [comment]
@pmac I think I found it in views.py :) Heavy python changes in this one, could use your R.

File: bedrock/base/templates/macros.html
Comment (alexgibson) [inline bedrock/base/templates/macros.html:57]: Is removing `data-spinner-color="{{ spinner_color }}` intentional here? Looks like it's still used by the JS? https://github.com/mozilla/bedrock/blob/master/media/js/base/send-to-device.js#L33

File: bedrock/base/templates/macros.html
Comment (alexgibson) [inline bedrock/base/templates/macros.html:89]: Might renaming this break something on the receiving end?

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx80.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx80.html:10]: Not entirely clear what this was used for, is it unrelated to this change?

File: tests/redirects/map_globalconf.py
Comment (alexgibson) [inline tests/redirects/map_globalconf.py:359]: Why is this test being removed?

**Review by alexgibson — CHANGES_REQUESTED** [review]
A few nits / questions, but otherwise the front-end changes lgtm. I'll defer to @pmac on the Python bits.

File: docs/send-to-device.rst
Comment (pmac) [inline docs/send-to-device.rst:81]: Are we sure we won't need the Geo Callback anymore? We do still sometimes restrict mobile products to certain countries.

**Review by pmac — CHANGES_REQUESTED** [review]
Just a couple of tweaks and questions about things that may not need to be removed. Nice work overall. 🍰

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx80.html
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx80.html:9]: Perhaps this was left over from an experiment? (that would be my guess anyway)

File: bedrock/base/templates/macros.html
Comment (stephaniehobson) [reply bedrock/base/templates/macros.html:89]: I think the change in `bedrock/firefox/views.py` is what was needed to avoid problems. Unless @pmac says different.

File: docs/send-to-device.rst
Comment (stephaniehobson) [reply docs/send-to-device.rst:81]: Right now the only restrictions we have are to only show the email form to locales which have translated emails and that's handled in the python. Since this is dead code now I'd err on the side of removing it but if there's a specific use case we know is coming I can save it.

**Comment by stephaniehobson** [comment]
Updated and pushed to integration tests again.

File: bedrock/firefox/views.py
Comment (pmac) [inline bedrock/firefox/views.py:205]: This line can be removed.

File: bedrock/firefox/views.py
Comment (pmac) [inline bedrock/firefox/views.py:208]: This form can also be cleaned up to remove the `number` field and the `clean_number` method.

https://github.com/mozilla/bedrock/blob/bbd6f42a9bf022097d83a1758cadec7f0c2fbf0a/bedrock/firefox/forms.py#L11

**Comment by stephaniehobson** [comment]
Updated, integration tests passing.

File: bedrock/base/templates/macros.html
Comment (alexgibson) [reply bedrock/base/templates/macros.html:89]: Seems ok to me, I tested the form end-to-end and got the email ok 👍

File: docs/send-to-device.rst
Comment (alexgibson) [reply docs/send-to-device.rst:81]: Agree, we can still make an XHR request to the geo URL for other reasons. I don't think it's worth keeping this in the S2D widget if it's functionality no longer related to the form 👍

**Review by alexgibson — APPROVED** [review]
Looks good to me. I'll hold off merging just in case @pmac wanted a last look at the most recent updates.

**Review by pmac — APPROVED** [review]
🍰

**Pull request merged**
